File: app/src/main/java/me/kirimin/mitsumine/ui/adapter/TagEditDialogFragmentAdapter.java
package me.kirimin.mitsumine.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;

import me.kirimin.mitsumine.R;
import rx.android.view.OnClickEvent;
import rx.android.view.ViewObservable;
import rx.functions.Action1;

public class TagEditDialogFragmentAdapter extends ArrayAdapter<String> {

    private final LayoutInflater inflater;
    private final Action1<OnClickEvent> onDeleteButtonClick;

    public TagEditDialogFragmentAdapter(Context context, ArrayList<String> tags, Action1<OnClickEvent> onDeleteButtonClick) {
        super(context, 0, tags);
        inflater = LayoutInflater.from(context);
        this.onDeleteButtonClick = onDeleteButtonClick;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.row_tag_edit_dialog_fragment, null);
            ViewHolder holder = new ViewHolder();
            holder.tagName = (TextView) convertView.findViewById(R.id.RegisterBookmarkTagEditDialogTagName);
            holder.deleteButton = (Button) convertView.findViewById(R.id.RegisterBookmarkTagEditDialogDelete);
            ViewObservable.clicks(holder.deleteButton).subscribe(onDeleteButtonClick);
            convertView.setTag(holder);
        }
        ViewHolder holder = (ViewHolder) convertView.getTag();
        holder.tagName.setText(getItem(position));
        holder.deleteButton.setTag(holder.tagName.getText().toString());
        return convertView;
    }

    public ArrayList<String> getAllItem() {
        ArrayList<String> tags = new ArrayList<>();
        for (int i = 0; i < getCount(); i++) {
            tags.add(getItem(i));
        }
        return tags;
    }

    static class ViewHolder {
        TextView tagName;
        Button deleteButton;
    }
}

File: app/src/main/java/me/kirimin/mitsumine/ui/adapter/TagEditDialogFragmentAdapter.kt
package me.kirimin.mitsumine.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.TextView

import java.util.ArrayList

import me.kirimin.mitsumine.R
import rx.android.view.OnClickEvent
import rx.android.view.ViewObservable
import rx.functions.Action1

public class TagEditDialogFragmentAdapter(context: Context, tags: ArrayList<String>, private val onDeleteButtonClick: Action1<OnClickEvent>) : ArrayAdapter<String>(context, 0, tags) {

    private val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View
        if (convertView == null) {
            view = inflater.inflate(R.layout.row_tag_edit_dialog_fragment, null)
            val holder = ViewHolder(
                    view.findViewById(R.id.RegisterBookmarkTagEditDialogTagName) as TextView,
                    view.findViewById(R.id.RegisterBookmarkTagEditDialogDelete) as Button)
            ViewObservable.clicks(holder.deleteButton).subscribe(onDeleteButtonClick)
            view.setTag(holder)
        } else {
            view = convertView
        }
        val holder = view.getTag() as ViewHolder
        holder.tagName.setText(getItem(position))
        holder.deleteButton.setTag(holder.tagName.getText().toString())
        return view
    }

    public fun getAllItem(): ArrayList<String> {
        val tags = ArrayList<String>()
        for (i in 0..getCount() - 1) {
            tags.add(getItem(i))
        }
        return tags
    }

    class ViewHolder(
            val tagName: TextView,
            val deleteButton: Button) {
    }
}
